torch/_refs/__init__.py

```diff
@@ -4971,8 +4971,14 @@ def is_finite(x):
         device=device,
         requires_grad=False,
     )
-    index = _maybe_convert_to_dtype(index, computation_dtype)
-    result = start + step * index
+
+    if integer_args:
+        result = index if step == 1 else step * index
+        result = result if start == 0 else start + result
+    else:
+        computation_dtype = utils.get_acc_type(dtype, device)
```

Comment:
As discussed, the fix is simply to do
```suggestion
        computation_dtype = torch.int64 if integer_args else utils.get_acc_type(dtype, device)
```